### PR TITLE
[V6] Deprecate geth miner namespace

### LINKED ISSUE #2856: Deprecate `miner` namespace support and tests
### What was wrong?

Geth removed support for their `miner` API since it's no longer needed after The Merge ™️ . See: https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-miner

### How can it be fixed?

Add deprecation warnings to the methods within the miner namespace, and change the tests to check for the deprecation warnings. We can then either remove the methods as Geth stops supporting these, or wait until v7 to remove.

### PR DESCRIPTION
**Pull request by reedsa**
### What was wrong?

Closes #2856

Geth removed support for their [miner API](https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-miner).

### How was it fixed?

Change `GethMiner` methods to `DeprecatedMethod`.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by pacrob — APPROVED** [review]
lgtm!

File: tests/integration/generate_fixtures/common.py [hunk 1]
```diff
@@ -210,7 +211,8 @@ def get_process(run_command):
 
 
 def mine_block(w3):
-    origin_block_number = w3.eth.block_number
+    with pytest.warns(DeprecationWarning):
```
Comment (kclowes) [inline tests/integration/generate_fixtures/common.py]: Do we need this here? It seems like this is just used for the fixture generation so you'd only see it if you were generating the fixture?

File: tests/integration/generate_fixtures/go_ethereum.py [hunk 2]
```diff
@@ -189,7 +190,8 @@ def verify_chain_state(w3, chain_data):
 
 
 def mine_block(w3):
-    origin_block_number = w3.eth.block_number
+    with pytest.warns(DeprecationWarning):
```
Comment (kclowes) [inline tests/integration/generate_fixtures/go_ethereum.py]: Same question here

**Review by kclowes — APPROVED** [review]
Looks good to me! I left a question around the `DeprecationWarning` in the fixture generation, but looks good otherwise.

**Comment by kclowes** [comment]
I also had to add `docs/web3.providers.rpc.rst` to the `exclude_patterns` in `docs/conf.py` but that is likely just me since the doc build is passing 🤷

File: web3/method.py [hunk 3]
```diff
@@ -214,17 +214,25 @@ def process_params(self, module: "Module", *args: Any, **kwargs: Any) -> Tuple[
...
 
     def __get__(
         self, obj: Optional["Module"] = None, obj_type: Optional[Type["Module"]] = None
     ) -> Any:
-        warnings.warn(
-            f"{self.old_name} is deprecated in favor of {self.new_name}",
-            category=DeprecationWarning,
-        )
+        if self.msg is not None:
+            warnings.warn(self.msg)
```
Comment (reedsa) [inline web3/method.py]: @kclowes should this include the `category=DeprecationWarning` argument as well?

**Pull request merged**